ydb/library/yql/core/cbo/cbo_optimizer_new.cpp

```diff
@@ -109,4 +109,123 @@ void TJoinOptimizerNode::Print(std::stringstream& stream, int ntabs) {
     RightArg->Print(stream, ntabs+1);
 }
 
+bool IsPKJoin(const TOptimizerStatistics& stats, const TVector<TString>& joinKeys) {
+    if (stats.KeyColumns.size()==0) {
+        return false;
+    }
+
+    for(size_t i=0; i<stats.KeyColumns.size(); i++){
+        if (std::find(joinKeys.begin(), joinKeys.end(), stats.KeyColumns[i]) == joinKeys.end()) {
+            return false;
+        }
+    }
+    return true;
+}
+
+bool TDummyProviderContext::IsJoinApplicable(const std::shared_ptr<IBaseOptimizerNode>& left,
```

Comment:
Lets rename TDummyProviderContext to something like TBaseProviderContext or TDefaultProviderContext.

Its not that dummy anymore if we're including the standard cardinality estimation in there

Reply:
Done.